README.md

```diff
@@ -108,43 +108,89 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Serialization is the process of converting an object into a stream of bytes to store the object in memory, a file or database, or transmit it through network. Its main purpose is to save the state of an object in order to be able to recreate it when needed. The reverse process is called deserialization.
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Hazelcast serializes all your objects into an instance of `Data`. `Data` is the binary representation of an object. When Hazelcast serializes an object into `Data`, it first checks whether the object is an instance of `IdentifiedDataSerializable`, if not it checks if it is an instance of `Portable` and serializes it accordingly. For the following types Hazelcast optimizes the serialization and you cannot override this behavior.
```

Comment:
"whether the object is an instance of `IdentifiedDataSerializable`" is not correct. IdentifiedDataSerializable object are not instances of IdentifiedDataSerializable. IdentifiedDataSerializable does not even exist in compiled code. What we check is if an object has the necessary methods to be serialized using a registered identified data serializer.

The same for portable part too.

Reply:
Doesn't `whether the object is an instance of IdentifiedDataSerializable` mean `whether the object implements necessary methods of IdentifiedDataSerializable` interface ?

Reply:
It is correct for typescript but we need to write this guide for Javascript users. Javascript does not have interfaces or implementing some stuff of another class. There is no class at all. I think it is important to be correct when using terms. I am not very sure how to explain this either. Better thing to do is to take a look at maybe other documentations or how js users generally explain this kind of relationship on the internet.